wallet/src/main/scala/org/bitcoins/wallet/internal/TransactionProcessing.scala

```diff
@@ -253,13 +265,16 @@ private[wallet] trait TransactionProcessing extends WalletLogger {
   /** If the given UTXO is marked as unspent, updates
     * its spending status. Otherwise returns `None`.
     */
-  private def markAsPendingSpent(
-      out: SpendingInfoDb): Future[Option[SpendingInfoDb]] = {
+  private def markAsSpent(
+      out: SpendingInfoDb,
+      spendingTxId: DoubleSha256DigestBE): Future[Option[SpendingInfoDb]] = {
     out.state match {
       case TxoState.ConfirmedReceived | TxoState.PendingConfirmationsReceived |
           TxoState.ImmatureCoinbase =>
```

Comment:
Does immature coinbase work here? 

I don't follow why we would want to transition from `TxoState.ImmatureCoinbase` -> `TxoState.PendingConfirmationsSpent` 

This is for the case where somebody pays us with the coinbase output right? Don't we want to transition the state this way 

1. ImmatureCoinbase
2. PendingConfirmationsReceived
3. ConfirmedReceived

and then whenever we spend the coinbase we go to `PendingConfirmationsSpent`?

Reply:
We don't allow creating a tx that spends an ImmatureCoinbase, however, if we do process one where we spend it, we should handle it correctly and mark it at spent rather than received. Otherwise we would count it as part of our balance once we have enough confirmations.

Reply:
Why don't we just return an exception (`Future.failed()`) as this seems to indicate wallet corruption? 

Is there any unit tests that test this behavior you mention above? If so could you link me?